src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -746,6 +757,13 @@ async Task OnStreamCanceled(InvocationRequest irq)
             _state.ReleaseConnectionLock();
         }
 
+        // The streaming method can continue after control is returned to the caller.
```

Comment:
InvokeAsync would have the same issue if you assigned the result task right?
```csharp
var invokeTask = connection.InvokeAsync<int>("Get");

// Activity.Current is now the invoke activity

await invokeTask;
```

Reply:
Maybe. I'll test.

Reply:
It's fine. I'm a little unsure how it does the right thing, but it's doing the right thing.